api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -27,10 +30,10 @@
   private static final int PROJECT_BILLING_ID_SIZE = 8;
   private static final Logger log = Logger.getLogger(BillingProjectBufferService.class.getName());
 
-  private final BillingProjectBufferEntryDao billingProjectBufferEntryDao;
-  private final Clock clock;
-  private final FireCloudService fireCloudService;
-  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+  private BillingProjectBufferEntryDao billingProjectBufferEntryDao;
```

Comment:
Why drop final here?

Reply:
I'll revert it back but for context, there is a bug with our version of Spring and App Engine where adding `@Transactional` to a method required a default constructor which is why I needed to drop `final`. It has something to do with its usage of Objenesis to create the proxy methods to support declarative transactions.